src/renderer/viz/expressions/basic/property.js

```diff
@@ -58,15 +58,29 @@ export default class Property extends BaseExpression {
         return feature[this.name];
     }
 
-    _bindMetadata (meta) {
-        const metaColumn = meta.properties[this.name];
+    get categories () {
+        return this.type === 'category'
+            ? this._categories
+            : [];
+    }
+
+    _bindMetadata (metadata) {
+        const metaColumn = metadata.properties[this.name];
         if (!metaColumn) {
             throw new Error(`Property '${this.name}' does not exist`);
         }
+
+        this._metadata = metadata;
+        this._categories = metaColumn.categories;
         this.type = metaColumn.type;
 
+        // FIXME
```

Comment:
Why FIXME?

Reply:
Because I am not very happy generating a getter in the constructor 😅but this can be addressed in other task, this one is pretty big already.